diff --git a/src/algorithms/instrumentation_operator.py b/src/algorithms/instrumentation_operator.py
index 86a7a74e..56f3336a 100644
--- a/src/algorithms/instrumentation_operator.py
+++ b/src/algorithms/instrumentation_operator.py
@@ -119,8 +119,9 @@ class InstrumentationOperator(TransformationAlgorithm):
             property = "NOOVERFLOW"
         try:
             subprocess.run(
-                [
-                    f"lib/cpachecker/bin/cpachecker",
+                [   "java", "-cp",
+                    f"{os.path.dirname(os.path.dirname(os.path.dirname(__file__)))}/cpachecker/runtime/*:{os.path.dirname(os.path.dirname(os.path.dirname(__file__)))}/cpachecker/cpachecker.jar",
+                    "org.sosy_lab.cpachecker.cmdline.CPAMain",
                     "--preprocess",
                     "--option",
                     "analysis.algorithm.instrumentation.instrumentationOperator=true",
